extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
 	//2. Add date range input field
 
 	const PAGINATION_SIZE = 50;
+	const REQUEST_SIZE = 200;
 	const DAYS_RANGE = 15;
 	const HTTP_STATUS_OK = 200;
-	const NO_USER_MATCH_ERROR = "Provided username did not match any user";
+	const NO_USER_MATCH_ERROR = 'discussionslog-no-user-match-error';
 
 	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
 
+	private $logger;
+	private $users;
+
 	public function __construct() {
 		parent::__construct( 'DiscussionsLog', '', false );
+		$this->users = [ ];
+		$this->logger = Wikia\Logger\WikiaLogger::instance();
 	}
 
 	public function index() {
 
-		if ( !$this->checkAccess() ) {
-			throw new \PermissionsError( 'specialdiscussionslog' );
-		}
+		$this->checkAccess();
 
 		$this->setHeaders();
+		$this->response->addAsset( '/extensions/wikia/SpecialDiscussionsLog/css/DiscussionsLog_Forms.scss' );
 
 		$this->wg->Out->setPageTitle( wfMessage( 'discussionslog-pagetitle' )->escaped() );
 
-		$userName = $this->wg->request->getVal( 'username', null );
+		list( $userName, $ipAddress ) = $this->readAndValidateInput();
```

Comment:
I think this could be simplified by making use of the `getParams()` method on the request object. That would get rid of this unpacking and then repacking we're doing of query params. I also don't think we have to check if there was a searchQuery before calling the userLog, the checks down in there make sure that nothing is rendered to the UI unless values were passed in.

I'm thinking something like this:
```PHP
...
    $this->validateInput();
    $this->setVal( 'inputForm', $this->sendSelfRequest( 'inputForm', $this->request->getParams() ) );
    $this->setVal( 'userLog', $this->sendSelfRequest( 'userLog', $this->request->getParams() );
}